### PR TITLE
feat(ollama): configurable probe/request timeouts via env vars

### LINKED ISSUE #4982: [Bug]: Ollama provider intermittently fails to register on startup — PROBE_TIMEOUT_MS=1500 too tight, exceptions silently swallowed
### Bug type
Regression / Incorrect behavior — intermittent silent failure

### Summary
On TUI startup, the Ollama provider intermittently fails to register because `isRunning()` uses a 1500 ms probe timeout that is exceeded under normal startup IO contention. The failure is silent: `.catch(() => {})` swallows all errors and the footer simply omits "Ollama".

### Steps to reproduce
1. `ollama serve` running locally on default port (`localhost:11434`) with at least one cloud model pulled (e.g. `kimi-k2.6:cloud`).
2. Quit any existing `gsd` process, then run `gsd` to start a fresh interactive TUI.
3. Repeat the quit/restart cycle ~5 times.
4. Observe: in some launches the footer shows `(ollama)` and `/model` lists Ollama models; in others it doesn't, with no error shown to the user.

### Expected behavior
Ollama provider registration should be deterministic when the daemon is healthy. If the probe is too slow it should retry or surface a warning, not silently drop the provider.

### Actual behavior
- Footer status `(ollama)` flickers between present/absent across cold restarts.
- `/model` selector is missing all Ollama models in the failing case.
- No user-visible error or warning. `~/.gsd/crash/` has no related entries.

### GSD version
`2.77.0`

### Node.js version
`v22.22.2`

### Operating system
Linux 6.17.0-22-generic (Ubuntu derivative)

### Install method
npm global install

### Affected area
AI provider / model integration

### Model and provider
`ollama/*` (any model — independent of which model is pulled)

### Logs, screenshots, and evidence

Diagnostic patch added to `dist/resources/extensions/ollama/index.js`'s `.then`/`.catch` callbacks (the only way to surface what the silent catch swallows). Two consecutive cold restarts:

```
# Failing case (intermittent)
[2026-04-25T14:44:12.190Z] hasUI=true probeAndRegister resolved: found=false OLLAMA_HOST=(default) OLLAMA_API_KEY_len=57

# Healthy case
[2026-04-25T14:46:06.044Z] [probe] start host=http://localhost:11434
[2026-04-25T14:46:07.926Z] [probe] isRunning=true took 71ms
[2026-04-25T14:46:08.996Z] [probe] discoverModels.length=7 took 1070ms
[2026-04-25T14:46:08.997Z] hasUI=true probeAndRegister resolved: found=true
```

The healthy `isRunning()` only takes 71 ms, but `dist/resources/extensions/ollama/ollama-client.js:4` hardcodes `PROBE_TIMEOUT_MS = 1500`. Under cold-start IO contention or a freshly woken Ollama daemon, fetch can exceed 1.5 s, hitting `AbortController.abort()` → `catch { return false }` → `probeAndRegister` returns false → `unregisterProvider("ollama")`.

The whole chain is observable only because I patched the silent `.catch(() => {})` in `index.js:113`. Without that, root cause is invisible to users.

Independent verification — running `dist/resources/extensions/ollama/ollama-discovery.js#discoverModels()` from the same shell with `localhost:11434/api/tags` returning HTTP 200 and 7 models always succeeds in standalone Node, confirming the failure is purely the 1.5 s probe-timeout race, not the daemon.

### Impact
- **Severity:** medium — Ollama is a configured provider but unusable until the user happens to win the race on a cold start.
- **Frequency:** intermittent, ~30–50% of cold restarts on a moderate-IO workstation.
- **Consequence:** users perceive Ollama as "broken" and can't easily diagnose because the failure is silent. Discoverability of the provider is essentially nondeterministic.

### Additional information

**Workaround (verified):** bump `PROBE_TIMEOUT_MS` from 1500 → 5000 in
`dist/resources/extensions/ollama/ollama-client.js`. With 5 s, healthy probes (~71 ms) still return instantly and IO-contended cold starts no longer abort prematurely. After this change Ollama registration is 100% reliable across 20+ test restarts.

**Suggested fixes (in order of preference):**
1. Raise `PROBE_TIMEOUT_MS` default to ~5000 ms — generous over the actual healthy probe time, still bounded.
2. Make it configurable via env var, e.g. `OLLAMA_PROBE_TIMEOUT_MS`, for users on slower hosts.
3. **Stop swallowing errors silently.** `dist/resources/extensions/ollama/index.js:113` `.catch(() => { ctx.ui.setStatus("ollama", undefined); })` should at minimum log to `~/.gsd/crash/` or `console.warn` when `process.env.GSD_DEBUG` is set, so future debugging doesn't require user-side patching.

**Related issues:**
- #3531 (closed) — covered the fallback resolver / `models.json` path, not the probe timeout.
- #4073 (closed) — handled the inverse direction (footer showing "Ollama" when not usable). Together with this bug, the TUI footer's reliability for Ollama remains imperfect.

### LINKED ISSUE #5003: Make Ollama probe and request timeouts configurable via env vars
## What

`src/resources/extensions/ollama/ollama-client.ts` uses two hard-coded timeouts:

```ts
const PROBE_TIMEOUT_MS = 1500;
const REQUEST_TIMEOUT_MS = 10000;
```

The 1500 ms startup probe is tight enough that several real-world setups hit it intermittently:

- Slow disk + contended IO when gsd2 boots alongside other heavy processes
- Remote Ollama over a high-latency link (e.g. ollama running on a LAN host or via tunnel)
- Cloud endpoints (`OLLAMA_HOST=https://ollama.com`) where TLS handshake alone can exceed 1.5 s on a cold connection

When the probe times out, the Ollama provider silently fails to register and the model picker shows no Ollama models — usually with no actionable diagnostic.

## Why an env var

`OLLAMA_HOST` is already configured this way, so this fits the existing surface area without introducing a new settings key. Defaults stay 1500 / 10000, so behavior is unchanged for everyone who doesn't set the var.

Current workarounds users have shared in the wild:
1. Patching the compiled `dist/.../ollama-client.js` (gets overwritten on every `npm run build`)
2. Maintaining a fork
3. Giving up and assuming Ollama is broken

All three are bad for an opt-in escape hatch.

## Proposed

Two new optional env vars:

| Var | Default | Purpose |
|---|---|---|
| `OLLAMA_PROBE_TIMEOUT_MS` | 1500 | Startup health probe |
| `OLLAMA_REQUEST_TIMEOUT_MS` | 10000 | REST API requests |

Both parsed via a shared helper that falls back gracefully on unset, empty, non-numeric, zero, or negative input — so a typo doesn't silently set the timeout to 0.

## Status

Have a working PR ready (~85 line diff including 10 unit tests covering the env-parsing edge cases). No behavior change unless either env var is set.

Happy to open the PR if this approach is acceptable. Or if you'd prefer a settings-based config (per-project preference instead of env), I can adapt — but env is consistent with `OLLAMA_HOST` and `OLLAMA_API_KEY` already in this same file.

### PR DESCRIPTION
**Pull request by jeremymcs**
## TL;DR

**What:** Adds `OLLAMA_PROBE_TIMEOUT_MS` and `OLLAMA_REQUEST_TIMEOUT_MS` env vars to override the hard-coded 1500 ms / 10 000 ms Ollama timeouts; also surfaces previously-silent probe failures under `GSD_DEBUG`.
**Why:** The 1500 ms startup probe intermittently times out on slower disks, LAN/remote Ollama, and cloud endpoints — Ollama silently disappears from the model picker with no diagnostic.
**How:** Defensive `envPositiveInt(name, fallback)` parser falls back on unset/empty/non-numeric/zero/negative input. `getProbeTimeoutMs()` / `getRequestTimeoutMs()` resolve at call time so changes are picked up live. Defaults unchanged (1500 / 10000).

## What

- `src/resources/extensions/ollama/ollama-client.ts` — replaces `PROBE_TIMEOUT_MS` / `REQUEST_TIMEOUT_MS` constants with `getProbeTimeoutMs()` / `getRequestTimeoutMs()`. New shared `envPositiveInt(name, fallback)` helper. All three exported for testability.
- `src/resources/extensions/ollama/index.ts` — replaces the `.catch(() => {})` silent swallows in the session-start probe path with a `debugOllama()` helper that writes a one-line diagnostic to stderr when `GSD_DEBUG` is set. Registration still soft-fails as before — only the diagnostic gets unblocked.
- `src/resources/extensions/ollama/tests/ollama-client-timeout-env.test.ts` (new) — 12 cases covering env-parsing fallbacks (unset, empty, `abc`, `0`, `-100`, valid integer, `1500ms`-style trailing junk) and live re-read on every call.

## Why

Closes #5003 (env var configurability), closes #4982 (probe timeout too tight + silent failures).

The 1500 ms startup probe is exceeded under several real-world setups:
- Slow disk + contended IO when GSD boots alongside other heavy processes.
- Remote Ollama over a high-latency link (LAN host, tunnel).
- Cloud endpoints (`OLLAMA_HOST=https://ollama.com`) where TLS handshake alone can exceed 1.5 s on a cold connection.

When the probe times out, the provider silently fails to register and the model picker shows no Ollama models. Reporters' workarounds today: patch `dist/` (overwritten on every build), maintain a fork, or assume Ollama is broken.

`OLLAMA_HOST` and `OLLAMA_API_KEY` already configure this same client via env, so this is a consistent surface area.

## How

The defensive parser:
```ts
function envPositiveInt(name, fallback) {
  const raw = process.env[name];
  if (!raw) return fallback;
  const parsed = Number.parseInt(raw, 10);
  if (!Number.isFinite(parsed) || parsed <= 0) return fallback;
  return parsed;
}
```

A typo like `OLLAMA_PROBE_TIMEOUT_MS=abc` or `=0` falls back to the documented default rather than silently disabling the timeout (`AbortController.abort()` with `0` ms = abort instantly).

Resolved at call time (not module load) so tests can mutate `process.env` between invocations and the production probe picks up overrides without a process restart.

The debug logging in index.ts:
```ts
const debugOllama = (where, error) => {
  if (process.env.GSD_DEBUG) {
    process.stderr.write(`[ollama] ${where} probe failed: ${msg}\n`);
  }
};
```

Default behaviour unchanged (no GSD_DEBUG → no output). This lets users self-diagnose silent failures without patching `dist/.../index.js`.

### Verification

- `npx tsc --noEmit -p tsconfig.extensions.json` — clean.
- `node --test ollama-client-timeout-env.test.ts ollama-client.test.ts` — 18/18 pass.

### Change type

- [x] `feat` — New env-var surface
- [x] `fix` — Surfaces previously-silent probe failures (closes #4982)

AI-assisted.

## Summary by CodeRabbit

* **New Features**
  * Environment variables added for configuring Ollama probe and request timeouts with sane defaults and clamping to safe limits.

* **Bug Fixes**
  * Probe failures no longer get silently discarded; errors are surfaced to debug output when debug mode is enabled while preserving soft-fail behavior.

* **Tests**
  * Unit tests added to validate env var parsing, fallback, clamping, and re-read behavior.

* **Documentation**
  * User docs updated with Ollama env vars, defaults, and timeout behavior.

### CODE REVIEW

**Comment by jeremymcs** [comment]
🤖 Dispatched `codex` with the following prompt:

<details>
<summary>Agent prompt (click to expand)</summary>

```text
You are acting as an autonomous PR babysitter for gsd-build/gsd-2 PR #5045.
PR URL: https://github.com/gsd-build/gsd-2/pull/5045
Base repository: gsd-build/gsd-2
Head repository: jeremymcs/gsd-2
Head branch: feat/5003-ollama-timeout-env
Head remote: fork-jeremymcs
You are running inside an isolated app-owned worktree.
Make only targeted changes that resolve the approved tasks.
Do not wait for user input, confirmation, or approval at any point.
Do not rewrite unrelated files.
Use the available git tooling in this environment.
GitHub follow-up replies and review-thread resolution will be handled by the babysitter after your run.
If a task is invalid after inspection, explain it in your final response and include the exact audit token.

Approved review-comment tasks:
None

Approved status-check tasks:
None

Approved documentation tasks:
Documentation updates are required for this PR.
Assessment summary: This PR adds user-facing Ollama configuration via OLLAMA_PROBE_TIMEOUT_MS and OLLAMA_REQUEST_TIMEOUT_MS. Repository configuration/operator docs should document the env vars, defaults, units, and fallback behavior.
Update the appropriate repository documentation for these changes.
Choose the right docs files for this repository (for example README, docs pages, API/config/operator docs).
If, after inspection, the repository documentation is already accurate or there is no appropriate docs target, leave docs unchanged and report that using the docs summary block with `no_change`.

When done:
1) Run the relevant verification for your changes.
2) If you changed files, commit and push to fork-jeremymcs HEAD:feat/5003-ollama-timeout-env.
3) For each feedback item you addressed or were blocked on, emit a summary block in the following format:
   FEEDBACK_SUMMARY_START <auditToken>
   <A concise 1-2 sentence summary of what you did or why you were blocked>
   FEEDBACK_SUMMARY_END
   Include one block per audit token. These summaries will be posted as follow-up comments on the PR.
4) If documentation tasks were assigned, emit exactly one docs summary block in the following format:
   DOCS_SUMMARY_START <changed|no_change>
   <A concise 1-2 sentence summary of the docs you updated, or why no docs changes were necessary after inspection>
   DOCS_SUMMARY_END
```

</details>

File: src/resources/extensions/ollama/ollama-client.ts [hunk 1]
```diff
@@ -19,8 +19,44 @@ import type {
...
+ * Defensive parsing: a typo like `OLLAMA_PROBE_TIMEOUT_MS=abc` or
+ * `OLLAMA_PROBE_TIMEOUT_MS=0` should not silently disable the timeout —
+ * fall back to the documented default instead.
+ */
+export function envPositiveInt(name: string, fallback: number): number {
+	const raw = process.env[name];
+	if (!raw) return fallback;
+	const parsed = Number.parseInt(raw, 10);
+	if (!Number.isFinite(parsed) || parsed <= 0) return fallback;
+	return parsed;
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/ollama/ollama-client.ts]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...
🧰 **Agent running** — `codex` is working on the fix...
✅ **Agent completed** — verifying changes...
🎉 **Resolved** — addressed in commit `d4ef437`.

**Comment by jeremymcs** [comment]
⏳ **Accepted** — this comment requires code changes. Queuing fix...
🧰 **Agent running** — `codex` is working on the fix...
✅ **Agent completed** — verifying changes...
🎉 **Resolved** — addressed in commit `d4ef437`.

**Comment by jeremymcs** [comment]
🤖 Dispatched `codex` with the following prompt:

<details>
<summary>Agent prompt (click to expand)</summary>

````text
You are acting as an autonomous PR babysitter for gsd-build/gsd-2 PR #5045.
PR URL: https://github.com/gsd-build/gsd-2/pull/5045
Base repository: gsd-build/gsd-2
Head repository: jeremymcs/gsd-2
Head branch: feat/5003-ollama-timeout-env
Head remote: fork-jeremymcs
You are running inside an isolated app-owned worktree.
Make only targeted changes that resolve the approved tasks.
Do not wait for user input, confirmation, or approval at any point.
Do not rewrite unrelated files.
Use the available git tooling in this environment.
GitHub follow-up replies and review-thread resolution will be handled by the babysitter after your run.
If a task is invalid after inspection, explain it in your final response and include the exact audit token.

Approved review-comment tasks:
1. [review_comment] coderabbitai[bot]
   file=src/resources/extensions/ollama/ollama-client.ts line=38
   sourceId=3144209507 sourceUrl=https://github.com/gsd-build/gsd-2/pull/5045#discussion_r3144209507
   replyKind=review_thread threadId=PRRT_kwDORj5hb859sPbX threadResolved=false
   auditToken=codefactory-feedback:gh-review-comment-3144209507
   _⚠️ Potential issue_ | _🟠 Major_

<details>
<summary>🧩 Analysis chain</summary>

🌐 Web query:

`In current Node.js documentation, how does setTimeout behave when delay is greater than 2147483647 milliseconds?`

💡 Result:

In the current Node.js documentation, when the delay parameter for setTimeout is greater than 2147483647 milliseconds (2^31 - 1, approximately 24.8 days), the delay is adjusted to 1 millisecond. Additionally, Node.js emits a TimeoutOverflowWarning indicating that the value does not fit into a 32-bit signed integer and the timeout duration was set to 1.

Citations:

- 1: https://nodejs.org/api/timers.html
- 2: https://nodejs.org/dist/latest-v22.x/docs/api/timers.html
- 3: https://github.com/nodejs/node/issues/55367
- 4: https://github.com/nodejs/node/commit/ce3586da31

---

🏁 Script executed:

```shell
# First, locate and examine the ollama-client.ts file
find . -name "ollama-client.ts" -type f
```

Repository: gsd-build/gsd-2

Length of output: 110

---

🏁 Script executed:

```shell
# Read the file to verify the code at lines 33-38 and understand the context
cat -n src/resources/extensions/ollama/ollama-client.ts | head -120
```

Repository: gsd-build/gsd-2

Length of output: 4888

---

🏁 Script executed:

```shell
# Check how envPositiveInt is used, particularly around line 98 with setTimeout
rg "setTimeout" src/resources/extensions/ollama/ollama-client.ts -B 3 -A 3
```

Repository: gsd-build/gsd-2

Length of output: 397

---



</details>

**Cap timeout env values to safe timer bounds to avoid accidental near-immediate aborts.**

Values from `envPositiveInt` flow through `getProbeTimeoutMs()` and `getRequestTimeoutMs()` to `setTimeout()` at line 98. Node.js clamps setTimeout delays exceeding 2,147,483,647 ms to 1 ms and emits a TimeoutOverflowWarning. An unvalidated large env value (typo, misconfiguration, or intentional test) will cause the timeout to fire almost immediately, aborting requests prematurely instead of using the intended delay.

<details>
<summary>🔧 Proposed fix</summary>

```diff
 const DEFAULT_HOST = "http://localhost:11434";
 const DEFAULT_PROBE_TIMEOUT_MS = 1500;
 const DEFAULT_REQUEST_TIMEOUT_MS = 10000;
+const MAX_TIMER_DELAY_MS = 2_147_483_647;

 export function envPositiveInt(name: string, fallback: number): number {
 	const raw = process.env[name];
 	if (!raw) return fallback;
 	const parsed = Number.parseInt(raw, 10);
-	if (!Number.isFinite(parsed) || parsed <= 0) return fallback;
-	return parsed;
+	if (!Number.isFinite(parsed) || parsed <= 0) return fallback;
+	return Math.min(parsed, MAX_TIMER_DELAY_MS);
 }
```
</details>

Add a test case verifying that env values above 2,147,483,647 are clamped to the limit.

<details>
<summary>🤖 Prompt for AI Agents</summary>

```
Verify each finding against the current code and only fix it if needed.

In `@src/resources/extensions/ollama/ollama-client.ts` around lines 33 - 38, The
envPositiveInt function currently returns any positive integer from process.env
and can produce values larger than Node's safe setTimeout limit; modify
envPositiveInt to clamp parsed values to a MAX_TIMEOUT_MS constant
(2_147_483_647) before returning so downstream callers like getProbeTimeoutMs
and getRequestTimeoutMs that feed setTimeout cannot trigger a
TimeoutOverflowWarning, and add a unit test that sets a very large env value and
asserts the returned value is capped at MAX_TIMEOUT_MS.
```

</details>

<!-- fingerprinting:phantom:poseidon:hawk:9e4089b7-5e04-4ad4-9bd4-83dc6c8f3065 -->

<!-- 4e71b3a2 -->

<!-- This is an auto-generated comment by CodeRabbit -->
2. [review] coderabbitai[bot]
   file=n/a line=n/a
   sourceId=4177495211 sourceUrl=https://github.com/gsd-build/gsd-2/pull/5045#pullrequestreview-4177495211
   replyKind=review threadId=n/a threadResolved=n/a
   auditToken=codefactory-feedback:gh-review-4177495211
   **Actionable comments posted: 1**

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
Verify each finding against the current code and only fix it if needed.

Inline comments:
In `@src/resources/extensions/ollama/ollama-client.ts`:
- Around line 33-38: The envPositiveInt function currently returns any positive
integer from process.env and can produce values larger than Node's safe
setTimeout limit; modify envPositiveInt to clamp parsed values to a
MAX_TIMEOUT_MS constant (2_147_483_647) before returning so downstream callers
like getProbeTimeoutMs and getRequestTimeoutMs that feed setTimeout cannot
trigger a TimeoutOverflowWarning, and add a unit test that sets a very large env
value and asserts the returned value is capped at MAX_TIMEOUT_MS.
```

</details>

<details>
<summary>🪄 Autofix (Beta)</summary>

Fix all unresolved CodeRabbit comments on this PR:

- [ ] <!-- {"checkboxId": "4b0d0e0a-96d7-4f10-b296-3a18ea78f0b9"} --> Push a commit to this branch (recommended)
- [ ] <!-- {"checkboxId": "ff5b1114-7d8c-49e6-8ac1-43f82af23a33"} --> Create a new PR with the fixes

</details>

---

<details>
<summary>ℹ️ Review info</summary>

<details>
<summary>⚙️ Run configuration</summary>

**Configuration used**: defaults

**Review profile**: CHILL

**Plan**: Pro Plus

**Run ID**: `9ba49d25-8b9f-4db0-9e3e-834e5ecba471`

</details>

<details>
<summary>📥 Commits</summary>

Reviewing files that changed from the base of the PR and between 84a383f51d88b6143f1c9331d9a26be1adacb08c and 6132d408928ab35aa3d7b61b255030c8aff11ed2.

</details>

<details>
<summary>📒 Files selected for processing (3)</summary>

* `src/resources/extensions/ollama/index.ts`
* `src/resources/extensions/ollama/ollama-client.ts`
* `src/resources/extensions/ollama/tests/ollama-client-timeout-env.test.ts`

</details>

</details>

<!-- This is an auto-generated comment by CodeRabbit for review status -->

Approved status-check tasks:
None

Approved documentation tasks:
Documentation updates are required for this PR.
Assessment summary: This PR adds user-facing Ollama configuration via OLLAMA_PROBE_TIMEOUT_MS and OLLAMA_REQUEST_TIMEOUT_MS. Repository configuration/operator docs should document the env vars, defaults, units, and fallback behavior.
Update the appropriate repository documentation for these changes.
Choose the right docs files for this repository (for example README, docs pages, API/config/operator docs).
If, after inspection, the repository documentation is already accurate or there is no appropriate docs target, leave docs unchanged and report that using the docs summary block with `no_change`.

When done:
1) Run the relevant verification for your changes.
2) If you changed files, commit and push to fork-jeremymcs HEAD:feat/5003-ollama-timeout-env.
3) For each feedback item you addressed or were blocked on, emit a summary block in the following format:
   FEEDBACK_SUMMARY_START <auditToken>
   <A concise 1-2 sentence summary of what you did or why you were blocked>
   FEEDBACK_SUMMARY_END
   Include one block per audit token. These summaries will be posted as follow-up comments on the PR.
4) If documentation tasks were assigned, emit exactly one docs summary block in the following format:
   DOCS_SUMMARY_START <changed|no_change>
   <A concise 1-2 sentence summary of the docs you updated, or why no docs changes were necessary after inspection>
   DOCS_SUMMARY_END
````

</details>

File: src/resources/extensions/ollama/ollama-client.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/ollama/ollama-client.ts]: Addressed in commit `d4ef437` by the latest babysitter run.

Clamped Ollama timeout environment values to Node's maximum safe timer delay and added a regression test for oversized values. This prevents large timeout misconfigurations from overflowing into near-immediate aborts.

**Comment by jeremymcs** [comment]
Addressed in commit `d4ef437` by the latest babysitter run.

> Responding to [comment by @coderabbitai[bot]](https://github.com/gsd-build/gsd-2/pull/5045#pullrequestreview-4177495211):
> **Actionable comments posted: 1**

Addressed the actionable Ollama timeout review item by bounding parsed positive env values and covering the cap behavior in the existing timeout env test suite.

File: src/resources/extensions/ollama/ollama-client.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/ollama/ollama-client.ts]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...
🧰 **Agent running** — `codex` is working on the fix...
✅ **Agent completed** — verifying changes...
🎉 **Resolved** — addressed in commit `6dcb873`.

**Comment by jeremymcs** [comment]
🤖 Dispatched `codex` with the following prompt:

<details>
<summary>Agent prompt (click to expand)</summary>

````text
You are acting as an autonomous PR babysitter for gsd-build/gsd-2 PR #5045.
PR URL: https://github.com/gsd-build/gsd-2/pull/5045
Base repository: gsd-build/gsd-2
Head repository: jeremymcs/gsd-2
Head branch: feat/5003-ollama-timeout-env
Head remote: fork-jeremymcs
You are running inside an isolated app-owned worktree.
Make only targeted changes that resolve the approved tasks.
Do not wait for user input, confirmation, or approval at any point.
Do not rewrite unrelated files.
Use the available git tooling in this environment.
GitHub follow-up replies and review-thread resolution will be handled by the babysitter after your run.
If a task is invalid after inspection, explain it in your final response and include the exact audit token.

Approved review-comment tasks:
1. [review_comment] coderabbitai[bot]
   file=src/resources/extensions/ollama/ollama-client.ts line=38
   sourceId=3144224784 sourceUrl=https://github.com/gsd-build/gsd-2/pull/5045#discussion_r3144224784
   replyKind=review_thread threadId=PRRT_kwDORj5hb859sPbX threadResolved=true
   auditToken=codefactory-feedback:gh-review-comment-3144224784
   `@jeremymcs`, here are the concrete changes needed to implement the fix:



## Task: Cap `envPositiveInt` to safe `setTimeout` bound

- Add `MAX_TIMER_DELAY_MS` constant to `ollama-client.ts`
- Clamp the parsed value in `envPositiveInt` using `Math.min`
- Add a unit test covering values above the limit

<!-- <requirement>
[requirement clamp_env_positive_int_1/2]
Add the `MAX_TIMER_DELAY_MS` constant and clamp the return value of `envPositiveInt` in `src/resources/extensions/ollama/ollama-client.ts`.

File: `src/resources/extensions/ollama/ollama-client.ts`

```diff
 const DEFAULT_HOST = "http://localhost:11434";
 const DEFAULT_PROBE_TIMEOUT_MS = 1500;
 const DEFAULT_REQUEST_TIMEOUT_MS = 10000;
+const MAX_TIMER_DELAY_MS = 2_147_483_647;

 export function envPositiveInt(name: string, fallback: number): number {
 	const raw = process.env[name];
 	if (!raw) return fallback;
 	const parsed = Number.parseInt(raw, 10);
 	if (!Number.isFinite(parsed) || parsed <= 0) return fallback;
-	return parsed;
+	return Math.min(parsed, MAX_TIMER_DELAY_MS);
 }
```

Also export `MAX_TIMER_DELAY_MS` so the test can reference it without duplicating the magic number:

```diff
-const MAX_TIMER_DELAY_MS = 2_147_483_647;
+export const MAX_TIMER_DELAY_MS = 2_147_483_647;
```
</requirement> -->

<!-- <requirement>
[requirement add_overflow_test_2/2]
Add a test case to `src/resources/extensions/ollama/tests/ollama-client-timeout-env.test.ts` asserting that a value above `MAX_TIMER_DELAY_MS` is clamped.

File: `src/resources/extensions/ollama/tests/ollama-client-timeout-env.test.ts`

Add the following test inside (or alongside) the existing `envPositiveInt` suite:

```typescript
import { envPositiveInt, MAX_TIMER_DELAY_MS } from "../ollama-client.js";

// inside the envPositiveInt describe block:
it("clamps values above MAX_TIMER_DELAY_MS to prevent setTimeout overflow", () => {
    withEnv("OLLAMA_PROBE_TIMEOUT_MS", String(MAX_TIMER_DELAY_MS + 1), () => {
        expect(envPositiveInt("OLLAMA_PROBE_TIMEOUT_MS", 1500)).toBe(MAX_TIMER_DELAY_MS);
    });
});

it("accepts MAX_TIMER_DELAY_MS exactly", () => {
    withEnv("OLLAMA_PROBE_TIMEOUT_MS", String(MAX_TIMER_DELAY_MS), () => {
        expect(envPositiveInt("OLLAMA_PROBE_TIMEOUT_MS", 1500)).toBe(MAX_TIMER_DELAY_MS);
    });
});
```
</requirement> -->

---

Failed to handle agent chat message. Please try again.



<!-- This is an auto-generated reply by CodeRabbit -->

Approved status-check tasks:
None

Approved documentation tasks:
None

When done:
1) Run the relevant verification for your changes.
2) If you changed files, commit and push to fork-jeremymcs HEAD:feat/5003-ollama-timeout-env.
3) For each feedback item you addressed or were blocked on, emit a summary block in the following format:
   FEEDBACK_SUMMARY_START <auditToken>
   <A concise 1-2 sentence summary of what you did or why you were blocked>
   FEEDBACK_SUMMARY_END
   Include one block per audit token. These summaries will be posted as follow-up comments on the PR.
4) If documentation tasks were assigned, emit exactly one docs summary block in the following format:
   DOCS_SUMMARY_START <changed|no_change>
   <A concise 1-2 sentence summary of the docs you updated, or why no docs changes were necessary after inspection>
   DOCS_SUMMARY_END
````

</details>

File: src/resources/extensions/ollama/ollama-client.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/ollama/ollama-client.ts]: Addressed in commit `6dcb873` by the latest babysitter run.

Confirmed `MAX_TIMER_DELAY_MS` is exported and `envPositiveInt` clamps to it, then added focused tests for `MAX_TIMER_DELAY_MS + 1` overflow clamping and accepting the exact max value.
